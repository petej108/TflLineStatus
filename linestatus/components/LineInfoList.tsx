import LineInfo from './LineInfo';
import { lineInfoType } from 'types/lineInfo';
import styles from '@/pages/index.module.css';

const LineInfoList = ({ linesData }: any) => {
  const getLineInfo = (lineInfoRaw: any): lineInfoType => {
    const lineInfo: lineInfoType = {
      lineId: lineInfoRaw.id as string,
      lineName: lineInfoRaw.name as string,
      lineStatus: lineInfoRaw.lineStatuses[0]
        .statusSeverityDescription as string,
    };
    return lineInfo;
  };

  const LineInfoListData: lineInfoType[] = linesData.map(getLineInfo);

  return (
    <>
      <div role='list' className={[styles.grid, styles.gridOuter].join(' ')}>
        {LineInfoListData.map((lineInfo) => (
          <LineInfo {...lineInfo} key={lineInfo.lineId} />
        ))}
      </div>
    </>
  );
};

export default LineInfoList;
