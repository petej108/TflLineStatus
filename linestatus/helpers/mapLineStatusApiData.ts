import { lineInfoType } from "@/types/lineInfoType";
import {lineStatusResponseType} from "@/types/lineStatusResponseType";


export const getLineInfoData = (lineStatusResponseData: any) => { 
    const getLineInfo = (lineInfoRaw: any): lineInfoType => {    
        const lineInfo: lineInfoType = {
          lineId: lineInfoRaw.id as string,
          lineName: lineInfoRaw.name as string,
          lineStatus: lineInfoRaw.lineStatuses[0]
            .statusSeverityDescription as string,
        };
        return lineInfo;
      };

      const LineInfoListData: lineInfoType[] =
        lineStatusResponseData.map(getLineInfo);
        return LineInfoListData;
};



 