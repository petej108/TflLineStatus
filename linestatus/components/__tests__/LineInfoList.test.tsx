import { render } from '@testing-library/react';
import LineInfoList from '@/components/LineInfoList';
import lineInfoApiDataMock from '../../data/lineStatusMock.json';
import { getLineInfoData } from '../../helpers/mapLineStatusApiData';

describe('Line Status List', () => {
  it('renders Line Info List unchanged', () => {
    const { container } = render(
      <LineInfoList linesData={getLineInfoData(lineInfoApiDataMock)} />
    );
    expect(container).toMatchSnapshot();
  });

  it('Line info List contains all items', () => {
    const { container } = render(
      <LineInfoList linesData={getLineInfoData(lineInfoApiDataMock)} />
    );

    expect(container.getElementsByClassName('card').length).toBe(
      lineInfoApiDataMock.length
    );
  });
});
