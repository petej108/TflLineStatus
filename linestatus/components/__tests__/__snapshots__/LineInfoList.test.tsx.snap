// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Status List renders Line Info List unchanged 1`] = `
<div>
  <div
    class="grid gridOuter"
    role="list"
  >
    <div
      class="card grid"
      id="bakerloo"
      role="listitem"
    >
      <div>
        Bakerloo
      </div>
       
      <div>
        Part Closure
      </div>
    </div>
    <div
      class="card grid"
      id="central"
      role="listitem"
    >
      <div>
        Central
      </div>
       
      <div>
        Good Service
      </div>
    </div>
    <div
      class="card grid"
      id="circle"
      role="listitem"
    >
      <div>
        Circle
      </div>
       
      <div>
        Severe Delays
      </div>
    </div>
    <div
      class="card grid"
      id="district"
      role="listitem"
    >
      <div>
        District
      </div>
       
      <div>
        Good Service
      </div>
    </div>
    <div
      class="card grid"
      id="hammersmith-city"
      role="listitem"
    >
      <div>
        Hammersmith & City
      </div>
       
      <div>
        Severe Delays
      </div>
    </div>
    <div
      class="card grid"
      id="jubilee"
      role="listitem"
    >
      <div>
        Jubilee
      </div>
       
      <div>
        Good Service
      </div>
    </div>
    <div
      class="card grid"
      id="metropolitan"
      role="listitem"
    >
      <div>
        Metropolitan
      </div>
       
      <div>
        Good Service
      </div>
    </div>
    <div
      class="card grid"
      id="northern"
      role="listitem"
    >
      <div>
        Northern
      </div>
       
      <div>
        Good Service
      </div>
    </div>
    <div
      class="card grid"
      id="piccadilly"
      role="listitem"
    >
      <div>
        Piccadilly
      </div>
       
      <div>
        Severe Delays
      </div>
    </div>
    <div
      class="card grid"
      id="victoria"
      role="listitem"
    >
      <div>
        Victoria
      </div>
       
      <div>
        Good Service
      </div>
    </div>
    <div
      class="card grid"
      id="waterloo-city"
      role="listitem"
    >
      <div>
        Waterloo & City
      </div>
       
      <div>
        Planned Closure
      </div>
    </div>
  </div>
</div>
`;
